client.go

```diff
@@ -377,10 +381,15 @@ func (s *Sender) send(ctx context.Context, msg *Message) (chan deliveryState, er
 		deliveryID     = atomic.AddUint32(&s.link.session.nextDeliveryID, 1)
 	)
 
-	// use uint64 encoded as []byte as deliveryTag
-	deliveryTag := make([]byte, 8)
-	binary.BigEndian.PutUint64(deliveryTag, s.nextDeliveryTag)
-	s.nextDeliveryTag++
+	var deliveryTag []byte
+	if msg.DeliveryTag == nil {
+		// use uint64 encoded as []byte as deliveryTag
+		deliveryTag = make([]byte, 8)
+		binary.BigEndian.PutUint64(deliveryTag, s.nextDeliveryTag)
+		s.nextDeliveryTag++
+	} else {
+		deliveryTag = msg.DeliveryTag
+	}
```

Comment:
Minor style thing, default to `msg.DeliveryTag` then check if the slice is empty, avoiding the else block.

``` go
deliveryTag := msg.DeliveryTag
if len(deliveryTag) == 0 {
    ...
}
```

Reply:
Much cleaner, will fix.